From 96640981be7b4d4e3f537d57324f7db3ef8c608f Mon Sep 17 00:00:00 2001
From: Raphael Mounier <mounierr07@gmail.com>
Date: Tue, 28 Mar 2023 08:07:10 +0200
Subject: [PATCH] Fix offline charger for Huawei phone

Change-Id: I76fdeb6ac845c1e12489778ba935f80260d75616

# Add ressource vendor charger init
# Add declaration of offlinecharger service
---
 rootdir/init.rc | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/rootdir/init.rc b/rootdir/init.rc
index a26e6679e..637c82c86 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -11,13 +11,18 @@ import /vendor/etc/init/hw/init.${ro.hardware}.rc
 import /system/etc/init/hw/init.usb.configfs.rc
 import /system/etc/init/hw/init.${ro.zygote}.rc
 
-service charger /bin/charger
+# Add vendor charger init
+import /vendor/etc/init/${ro.bootmode}/init.${ro.bootmode}.rc
+
+# Add offlinecharger service from Huawei
+service charger /system/bin/offlinecharger -p
     class charger
-    user system
-    group system shell graphics input wakelock
+    user root
+    group root system shell graphics input wakelock
     capabilities SYS_BOOT
     seclabel u:r:charger:s0
 
+
 on boot
     chown system system /sys/class/power_supply/batery/input_suspend
     chmod 0777 /sys/class/power_supply/batery/input_suspend
-- 
2.25.1

